programs/bpf/c/src/invoke/invoke.c

```diff
@@ -251,9 +251,9 @@ extern uint64_t entrypoint(const uint8_t *input) {
     sol_assert(SUCCESS ==
                sol_invoke(&instruction, accounts, SOL_ARRAY_SIZE(accounts)));
 
+    // Signer privilege escalation will always fail the whole transaction
     instruction.accounts[0].is_signer = true;
-    sol_assert(SUCCESS !=
-               sol_invoke(&instruction, accounts, SOL_ARRAY_SIZE(accounts)));
+    sol_invoke(&instruction, accounts, SOL_ARRAY_SIZE(accounts));
```

Comment:
Why was the assert removed?

Reply:
Ah, moved out to `tests/program.rs`?

Reply:
Not exactly. The assert wasn't doing anything because the invalid invoked instruction causes an error in message processing and control never returns to the caller, as I understand it.